src/MPI.jl

```diff
@@ -33,14 +33,49 @@ function _doc_external(fname)
 """
 end
 
-try
-    include(joinpath(dirname(@__DIR__), "deps", "deps.jl"))
-    include(joinpath(dirname(@__DIR__), "deps", "compile_time_mpi_constants.jl"))
-catch e
-    error("MPI.jl not properly configured, please run `Pkg.build(\"MPI\")`.")
+
+import MPIPreferences
+
+@static if MPIPreferences.binary == "MPICH_jll"
```

Comment:
My understanding is that a top-level `@static` is redundant, the expression is executed at precompile time anyway.

Reply:
Indeed!